change FileStore to use FilesystemManager with disk configuration

stevethomas opened the pull request:
This PR:
- swaps `Illuminate\Filesystem\Filesystem` for `Illuminate\Filesystem\FilesystemManager`
- adds a config item `credential_disk` which defaults to the default Laravel disk
- updates the readme

With this change we are changing to the default configured filesystem, making it trivial to access xero.json from a distributed disk like S3, rather than the previous hardcoding to a local disk. 

This solves the problem of accessing the credential store when multiple servers are in-use, or in a serverless environment like Lambda / Vapor.

hailwood reviewed (changes requested):
Thanks for sending the PR through, you got onto that super quick!

It's looking good, just a few things I'd like changed on how we access the disk due to changes in L9 :)

stevethomas commented:
Good catch on the changing `FILESYSTEM_DISK` default. I implemented `XERO_CREDENTIAL_DISK` as suggested.  

I have implemented the other changes as well. I dropped the `$files` property as it was redundant. 

Another idea if you are tagging a new release; it might be a good opportunity to drop Laravel 5.5 (at least).

hailwood commented:
Thanks @stevethomas!

v3.0.0 has just been released :)
